src/CodeGen_Hexagon.cpp

```diff
@@ -50,6 +50,9 @@ CodeGen_Hexagon::CodeGen_Hexagon(Target t)
     user_assert(!target.features_all_of(
         {Halide::Target::HVX_128, Halide::Target::HVX_64}))
         << "Cannot set both HVX_64 and HVX_128 at the same time.\n";
+    user_assert(target.features_any_of(
```

Comment:
I think this assert should go inside `acquire_hvx_context`, so we only hit this assert when we actually ask for HVX (which doesn't necessarily always happen).

Reply:
OK, but not sure I understand -- how/why would we create a Codegen_Hexagon if we weren't going to actually do HVX codegen?

Reply:
You can run code on Hexagon without HVX. It's pretty theoretical, but someone might try to do it (e.g. a small glue pipeline between larger pipelines that doesn't need HVX).

I just really don't think asserts should be more strict than necessary.

Reply:
Aah good point. Thanks for catching this @dsharletg